src/django/api/views.py

```diff
@@ -1211,3 +1214,52 @@ def api_feature_flags(request):
     }
 
     return Response(response_data)
+
+
+@transaction.atomic
+@api_view(['POST'])
+@permission_classes((IsRegisteredAndConfirmed,))
+def claim_facility(request, oar_id=None):
```

Comment:
I made an api_view method for this because I wanted to use the permission classes decorator, which wasn't possible with the class method. You should also try to CURL at this endpoint and verify that it returns an authorization error.

Reply:
On a hunch I tried passing `permission_classes` to the `@action` decorator on the facilities `count` method and it appears to work correctly.

```diff
diff --git a/src/django/api/views.py b/src/django/api/views.py
index ac3afde..a058ba9 100644
--- a/src/django/api/views.py
+++ b/src/django/api/views.py
@@ -533,7 +533,8 @@ class FacilitiesViewSet(ReadOnlyModelViewSet):
         except Facility.DoesNotExist:
             raise NotFound()
 
-    @action(detail=False, methods=['get'])
+    @action(detail=False, methods=['get'],
+            permission_classes=(IsRegisteredAndConfirmed,))
     def count(self, request):
         """
         Returns a count of total Facilities available in the Open Apparel

```

I think we can use this to nest an authenticated `claim` action under the otherwise open `FacilitiesViewSet`.

Reply:
Great! I'll update this and drop the newly added URL pattern.